drivers/SmartThings/philips-hue/src/handlers.lua

```diff
@@ -247,6 +248,7 @@ end
 ---@param light_device HueChildDevice
 local function do_refresh_light(driver, light_device)
```

Comment:
This function has become exceedingly smelly, but I didn't think that now was the right time to be dealing with that. One of the major issues is that the whole driver is oriented around light devices on our platform representing a light resource in the Hue representation. We should probably revise that to orient around the Hue device representation instead of the light resource representation, but that's a deep surgery on the entire driver.